lib/worker.js

```diff
@@ -150,6 +155,23 @@ async function updateDependency(upgrade) {
       newVersion);
   }
 
+  function getYarnLockFileName(packageFileName) {
+    const packagePath = packageFileName.split('/');
+    // Remove 'package.json'
+    const basePath = packagePath.slice(0, -1);
+    basePath.push('yarn.lock');
+    return basePath.join('/');
+  }
+
+  function generateLockFile(newPackageJson) {
+    logger.debug('Generating new yarn.lock file');
+    rimraf.sync('/tmp/renovate-yarn');
+    mkdirp.sync('/tmp/renovate-yarn');
+    fs.writeFileSync('/tmp/renovate-yarn/package.json', newPackageJson);
+    cp.spawnSync('yarn', ['install'], { cwd: '/tmp/renovate-yarn' });
```

Comment:
@rarkins your implementation is what I understood @jonbretman's comment to mean. There are projects that have project-specific `.npmrc` files to configure the location of registries containing scoped packages.

As for whether those `.npmrc` files may contain an auth token. they may. `renovate`, though, is run by the owner of the project in question. In my mind that contains the issue (it's not like a third party is downloading my auth tokens onto their servers).